lib/enum/basic_enum.rb

```diff
@@ -85,66 +102,83 @@ def element_frequency_count
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-find-all
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-select
   def elements_ending_in_er
+    collection.find_all{ |word| word.include? "er"}
   end
 
 
   # Refer to find_index
   # http://ruby-doc.org/core-2.1.4/Enumerable.html#method-i-find-index
   def index_of_first_awesome_element
+    collection.find_index{ |word| word[:awesome]}# same question as finding the first awesome person... how does word relate to [:awesome]
```

Comment:
Same as before - the collection is made of people, so your local variable should be person, not word. Then, you're looking up the value of [awesome] for each person, like this: `person[:awesome]`. The method find_index will return the index of the person where `person[:awesome]` returns `true`.